flink-end-to-end-tests/flink-end-to-end-tests-hbase/src/main/java/org/apache/flink/tests/util/hbase/LocalStandaloneHBaseResource.java

```diff
@@ -92,62 +95,98 @@ private void setupHBaseDist() throws IOException {
 	}
 
 	private void setupHBaseCluster() throws IOException {
-		LOG.info("Starting HBase cluster");
-		AutoClosableProcess.runBlocking(
-			hbaseDir.resolve(Paths.get("bin", "start-hbase.sh")).toString());
-
-		while (!isHBaseRunning()) {
-			try {
-				LOG.info("Waiting for HBase to start");
-				Thread.sleep(500L);
-			} catch (InterruptedException e) {
-				Thread.currentThread().interrupt();
-				break;
-			}
-		}
+		LOG.info("Starting HBase cluster...");
+		runHBaseProcessWithRetry("start-hbase.sh", () -> !isHMasterRunning());
+		LOG.info("Start HBase cluster success");
 	}
 
 	@Override
 	public void afterTestSuccess() {
+		shutdownResource();
+		downloadCache.afterTestSuccess();
+		tmp.delete();
+	}
+
+	private void shutdownResource() {
+		LOG.info("Stopping HBase Cluster...");
 		try {
-			LOG.info("Stopping HBase Cluster");
-			AutoClosableProcess.runBlocking(
-				hbaseDir.resolve(Paths.get("bin", "hbase-daemon.sh")).toString(),
-				"stop",
-				"master");
+			runHBaseProcessWithRetry("stop-hbase.sh", () -> isHMasterAlive());
```

Comment:
I'd prefer to shutdown the hbase process by killing the process id because I encountered the hung case before when calling the `stop-hbase.sh`.